addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,333 @@
+/**
+ * @author RT_Borg
+ *
+ * Uses topic id as a proxy for age, which doesn't require loading other pages
+ * to read actual dates.
+ */
+
+export default async function ({ addon, global, console, msg }) {
+  // Default is a little over a month's worth of topics. Reasonably stable through 2022.
+  // There's a lot of tolerance. Half or twice as many topics filter pretty much the same.
+  const defaultTopicInterval = 7500;
+  const defaultForums = [
+    "New Scratchers",
+    "Help with Scripts",
+    "Questions about Scratch",
+    "Bugs and Glitches",
+    "Project Ideas",
+  ];
+  // Forums beyond these seem very poor options,
+  // as many posts live long or are appropriately revived
+  const customForumSettingToForum = {
+    applyToNewScratchers: "New Scratchers",
+    applyToHelpWithScripts: "Help with Scripts",
+    applyToQuestionsAboutScratch: "Questions about Scratch",
+    applyToBugsAndGlitches: "Bugs and Glitches",
+    applyToProjectIdeas: "Project Ideas",
+    applyToRequests: "Requests",
+    applyToSuggestions: "Suggestions",
+    applyToOpenSourceProjects: "Open Source Projects",
+  };
+  const searchResultsPageName = "Search Results"; // recent posts and unanswered posts
+  // mobile urls look like: https://scratch.mit.edu/discuss/m/6/*
+  const isMobileSite = /^\/discuss\/m\//.test(location.pathname);
+  // tableBodyNode will be null on mobile and non-table search page:
+  // https://scratch.mit.edu/discuss/search/?action=show_user&show_as=posts
+  const tableBodyNode = document.querySelector("tbody"); // null on mobile
+
+  if (!tableBodyNode && !isMobileSite) {
+    // Standard site page without a table. Don't run on this page.
+    return;
+  }
+
+  const forumOrSearchPageName = await determineForumOrSearchPageName();
+
+  /**
+   * An array of topic
+   * Each topic is:
+   *    - topicId      int, used to compare topic "age" relative to other posts on the page
+   *    - topicCell    Node that will be highlighted if too old
+   *    - forum        String, usually the forum being listed. Comes from the row on a search page
+   *    - restoreCell  Only defined if the cell is modified. A clone of the initial state
+   *                   of the cell Node, to revert for disable addon reload and settings changes
+   */
+  let topics = [];
+
+  highlightNecropostsIfOnSelectedForumPage();
+
+  addon.settings.addEventListener("change", onAddonSettingsChange);
+  addon.self.addEventListener("disabled", onDisabled);
+  addon.self.addEventListener("reenabled", onReenabled);
```

Comment:
Could the referenced functions here be used as arrow functions?

Reply:
If I understand the references I'm reading about => correctly, the different closures wouldn't be relevant in these cases. (I'd have to test it.) To me, this code looks easy to understand, but maybe that's because I'm used to idioms from other languages.

Are you pointing out a correctness issue, or raising a style preference?

Working on bugs in the infinite-scroll addon, forumScroll.js, I see that instead of 
```
function update () {...}
window.addEventListener("scroll", update, {passive: true});
```
the code is
https://github.com/ScratchAddons/ScratchAddons/blob/d1a7bbf61ec2cfb8ca470138dab528311452227b/addons/infinite-scroll/forumScroll.js#L19
https://github.com/ScratchAddons/ScratchAddons/blob/d1a7bbf61ec2cfb8ca470138dab528311452227b/addons/infinite-scroll/forumScroll.js#L66

Is that what you're suggesting? It looks awkward to me, but if that's idiom Scratch Addons uses, I can conform. Please let me know if I need to make these changes.

Reply:
> Is that what you're suggesting? It looks awkward to me, but if that's the idiom Scratch Addons uses, I can conform. Please let me know if I need to make these changes.

I'm suggesting something like this:
```javascript
addon.self.addEventListener("disabled", () => {
	// ...
});
```

Reply:
I agree with Weredime. It doesn't necessarily have to be an arrow function but could you put the dynamic enable/disable code here instead of using separate functions?